Blog/templates/Blog/post_list.html

```diff
@@ -2,27 +2,73 @@
+            <a href="{% url 'posts' %}" class="button">No Tag</a>
+            {% if tags %}
+                {% for tag in tags %}
+                    <a href="{% url 'posts' %}?tag={{ tag }}" class="button">{{ tag }}</a>
+                {% endfor %}
+            {% else %}
+                <p>No Tags available...</p>
+            {% endif %}
+        </div>
+    </div>
 
-    {% if post_list %}
+    {% if object_list %}
         <div class="row">
-            {% for post in post_list %}
-                <div class="card cell-lg-5 offset-lg-1 cell-sd-full">
-                    <div class="card-header">
-                        <h2>{{ post.title }}</h2>
-                    </div>
-                    <div class="card-content p-2">
-                        <p>{{ post.content|truncatechars:240 }}</p>
-                    </div>
-                    <div class="card-footer">
-                        <a class="button secondary" href="{{ post.get_absolute_url }}">Read More</a>
-                    </div>
+        {% for post in object_list %}
+            <div class="card cell-lg-5 offset-lg-1 cell-sd-full">
+                <div class="card-header">
+                    <h2>{{ post.title }}</h2>
+                </div>
+                <div class="card-content p-2">
+                    <p>{{ post.content|truncatechars:240 }}</p>
                 </div>
-            {% endfor %}
+                <div class="card-footer">
+                    <a class="button secondary" href="{{ post.get_absolute_url }}">Read More</a>
+                </div>
+            </div>
+        {% endfor %}
 
-        </div>
     {% else %}
         <p>There are no posts available..</p>
     {% endif %}
+{% if post_list.has_other_pages %}
+    <div class="offset-4">
+
+        <ul class="pagination secondary no-gap">
+
+            {% if post_list.has_previous %}
+                <li class="page-item"><a class="page-link" href="?page={{ post_list.previous_page_number }}">&laquo;</a>
+                </li>
+            {% else %}
+                <li class="page-item"><span class="page-link">&laquo;</span></li>
+            {% endif %}
+
+            {% if pagination_range %}
+                {% for i in pagination_range %}
+                    {% if i == post_list.number %}
+                        <li class="page-item active"><a class="page-link" href="?page={{ i }}">{{ i }}</a></li>
+                    {% else %}
+                        <li class="page-item"><a class="page-link" href="?page={{ i }}">{{ i }}</a></li>
+                    {% endif %}
+                {% endfor %}
+            {% endif %}
 
+            {% if post_list.has_next %}
+                <li class="page-item"><a class="page-link" href="?page={{ post_list.next_page_number }}">&raquo;</a>
+                </li>
+            {% else %}
+                <li class="page-item"><span class="page-link">&raquo;</span></li>
+            {% endif %}
+        </ul>
+    </div>
+{% endif %}
+</div>
```

Comment:
It seems like the opening div tag is defined inside the `{% if %}` block. See my other comment about the indentation.